poc/OwinTestApp/src/OwinClassLibrary/Zipkin/ZipkinMessageHandler.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System.Diagnostics;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using OwinClassLibrary.Cat;
+
+namespace OwinClassLibrary.Zipkin
+{
+    public class ZipkinMessageHandler : DelegatingHandler
+    {
+        private readonly ITraceProvider _provider;
+
+        public ZipkinMessageHandler(ITraceProvider provider)
+            : this(provider, new HttpClientHandler())
+        {
+        }
+
+        public ZipkinMessageHandler(ITraceProvider provider, HttpMessageHandler innerHandler)
+            : base(innerHandler)
+        {
+            _provider = provider;
+        }
+
+        protected override Task<HttpResponseMessage> SendAsync(
+            HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            var nextTrace = _provider.GetNext();
```

Comment:
call `GetNext` here since the value is used in request header and zipkin.StartClientSpan().